java/arcs/core/allocator/Allocator.kt

```diff
@@ -275,13 +267,20 @@ class Allocator private constructor(
             RamDiskStorageKey("partitions")
         )
 
-        // TODO(b/152435365) - Transition to using entity handle
         suspend fun create(
             hostRegistry: HostRegistry,
-            handleManager: HandleManager
+            handleManager: EntityHandleManager
         ): Allocator {
-            val collection = handleManager.rawEntityCollectionHandle(STORAGE_KEY, SCHEMA)
-            return Allocator(hostRegistry, collection)
+            val collection = handleManager.createCollectionHandle(
+                HandleMode.ReadWrite,
+                "partitions",
+                EntityBaseSpec(SCHEMA),
+                STORAGE_KEY
+            )
+            return Allocator(
+                hostRegistry,
+                collection as ReadWriteCollectionHandleAdapter<EntityBase>
```

Comment:
If this gives you an unchecked cast warning, please add @Suppress("UNCHECKED_CAST") annotation to the method